modules/oe_subscriptions_digest/src/SubscriptionsFormAlter.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_subscriptions_digest;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class to manage digest preferences.
+ *
+ * @internal
+ */
+final class SubscriptionsFormAlter implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The user account for which the form is being rendered.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected UserInterface $account;
+
+  /**
+   * Creates a new instance of this class.
+   *
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $moduleHandler
+   *   The module handler.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(protected ModuleHandlerInterface $moduleHandler, protected EntityTypeManagerInterface $entityTypeManager) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('module_handler'),
+      $container->get('entity_type.manager'),
+    );
+  }
+
+  /**
+   * Alters the form to add digest preference.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function subscriptionsFormAlter(array &$form, $form_state): void {
+    $this->moduleHandler->loadInclude('message_digest_ui', 'module');
+    $this->account = $form_state->getBuildInfo()['args']['0'];
```

Comment:
The main issue with this is that the class will be stored in the form state as a whole, including a serialised account variable and the services. We can solve this by using \Drupal\Core\DependencyInjection\DependencySerializationTrait , but not for the user.
I usually prefer to avoid referencing the class in a callback, and instead use a static callback.
This means that we need to reload the user from the form state in the submit, but it's not a big deal, we can make a protected static method.

Reply:
Added the static method and removed the property